src/services/iiif-parser.js

```diff
@@ -373,12 +368,10 @@ export function getCustomStart(manifest) {
   let startProp = parseManifest(manifest).getProperty('start');
 
   let getCanvasIndex = (canvasId) => {
-    const canvasIds = canvasesInManifest(manifest);
-    const currentCanvasIndex = canvasIds
-      .map(function (c) {
-        return c;
-      })
-      .indexOf(canvasId);
+    const currentCanvasIndex = canvasesInManifest(manifest)
+      .findIndex((c) => {
+        return c.canvasId === canvasId;
+      });
     return currentCanvasIndex;
   };
```

Comment:
This pattern seems to be coming up a few times.  Maybe it could be extracted into a helper function in a future PR?

Reply:
Yes :100: That sounds good. I'll make a note.